packages/block-library/src/image/view.js

```diff
@@ -355,6 +443,24 @@ const { state, actions, callbacks } = store(
 				}
 			`;
 			},
+			setScreenReaderText() {
+				const { ref } = getElement();
+				if ( ! state.overlayEnabled ) {
+					ref.textContent = '';
+				} else if ( state.images.length === 1 ) {
+					ref.textContent = state.currentImage.alt
+						? `Enlarged image: ${ state.currentImage.alt }`
+						: 'Enlarged image';
+				} else {
+					ref.textContent = state.currentImage.alt
+						? `Enlarged image ${ state.currentImageIndex + 1 } of ${
+								state.images.length
+						  }: ${ state.currentImage.alt }`
+						: `Enlarged image ${ state.currentImageIndex + 1 } of ${
+								state.images.length
+						  }`;
```

Comment:
What is the plan to make these messages translatable? That's a blocker for this PR.

Reply:
The plan is to move the population of the gallery array to the server, similar to how it was done in the accordion block [in this commit](https://github.com/WordPress/gutenberg/pull/64119/commits/1b83689ac1f4fd312604c3ca9a8657a3a13b1911), and use the array length to add the translated texts on PHP to the metadata of each image.

Reply:
Right now I don't know how to do that 😅

@luisherranz As per [this document](https://developer.wordpress.org/block-editor/reference-guides/interactivity-api/core-concepts/server-side-rendering/#serializing-other-processed-values-to-be-consumed-on-the-client), is it possible to process and translate strings on the server and send them to the client?

Reply:
@madhusudhand, are you still planning to implement this change?

Reply:
fixed in b9443550f140e44df1dab807c71becc047806bc7